src/Makefile.am

```diff
@@ -728,14 +701,60 @@ libbitcoin_common_a_SOURCES += common/url.cpp
 endif
 #
 
+# kernel #
+libbitcoin_kernel_a_CPPFLAGS = $(AM_CPPFLAGS) $(BITCOIN_INCLUDES) $(BOOST_CPPFLAGS) $(LEVELDB_CPPFLAGS)
+libbitcoin_kernel_a_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS)
+libbitcoin_kernel_a_SOURCES = \
+  chain.cpp \
+  compressor.cpp \
+  consensus/tx_verify.cpp \
+  dbwrapper.cpp \
+  deploymentinfo.cpp \
+  deploymentstatus.cpp \
+  flatfile.cpp \
+  kernel/chain.cpp \
+  kernel/checks.cpp \
+  kernel/coinstats.cpp \
+  kernel/context.cpp \
+  kernel/cs_main.cpp \
+  kernel/disconnected_transactions.cpp \
+  kernel/mempool_persist.cpp \
+  kernel/mempool_removal_reason.cpp \
+  node/blockstorage.cpp \
+  node/chainstate.cpp \
+  node/utxo_snapshot.cpp \
+  policy/fees.cpp \
+  policy/packages.cpp \
+  policy/rbf.cpp \
+  policy/settings.cpp \
+  pow.cpp \
+  script/sigcache.cpp \
+  signet.cpp \
+  txdb.cpp \
+  txmempool.cpp \
+  validation.cpp \
+  validationinterface.cpp \
+  versionbits.cpp \
+  warnings.cpp \
+  $(BITCOIN_CORE_H)
+#
+
 # util #
 libbitcoin_util_a_CPPFLAGS = $(AM_CPPFLAGS) $(BITCOIN_INCLUDES)
 libbitcoin_util_a_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS)
 libbitcoin_util_a_SOURCES = \
   clientversion.cpp \
+  coins.cpp \
```

Comment:
In commit "build: Introduce libbitcoin_kernel as an internal compilation unit" (1f98ed5e56230a61b699f669459e39fa101a1ca5)

It's surprising to `coins.cpp` moving from common library to the util library instead kernel library. Curious if this is necessary.

It might be nice to explain general reasoning behind moves in the commit description and call out any special cases.

I'm guessing that certain of these files that are moved to the util library like `coins.cpp` and `kernel/chainparams.cpp` probably could be split up later and mostly moved back to the kernel, since wallet and gui code shouldn't need most of what is in them.